docs/proposals/15-manage-bastions-and-ssh-key-pair-rotation.md

```diff
@@ -0,0 +1,201 @@
+
+### SSH Flow
+0. Users should only get the RBAC permission to `create` / `update` `Bastion` resources for a namespace, if they should be allowed to SSH onto the shoot nodes in this namespace.
+1. User/`gardenctlv2` creates `Bastion` resource in garden cluster (see resource example below)
+    - First, gardenctl would figure out the external IP. Either by calling an external service (gardenctl (v1) uses https://github.com/gardener/gardenctl/blob/master/pkg/cmd/miscellaneous.go#L226) or by calling a binary that prints the external IP(s) to stdt out. The binary should be configurable. The result is set under `spec.clientIP`
+    - the public PGP key of the user is set under `spec.publicKey`. The key that should be used needs to be configured beforehand by the user
+    - The targeted shoot is set under `spec.shootRef`
+2. GAPI Admission Control for the `Bastion` resource in the garden cluster
+    - Mutating Webhook
+        - according to shootRef, sets the `spec.seedName`
+        - according to shootRef, sets the `spec.providerType`
+        - on creation, sets `metadata.annotations["operations.gardener.cloud/created-by"]` according to the user that created the resource
+    - Validating Webhook for the `Bastion` resource
+        - For security reasons, it validates that only the user who created the resource can update the spec so that for example another user cannot sneak in his own ip for the bastion firewall rule and own public PGP key, to be able to decrypt the SSH private key
+3. `gardenlet`
+    - Watches `Bastion` resource for own seed under api group `operations.gardener.cloud` in the garden cluster
+    - Creates `Bastion` custom resource under api group `extensions.gardener.cloud/v1alpha1` in the seed cluster
+4. Gardener extension provider <infra> / Bastion Controller on Seed:
+    - With own `Bastion` Custom Resource Definition in the seed under the api group `extensions.gardener.cloud/v1alpha1`
+    - Watches `Bastion` custom resources that are created by the `gardenlet` in the seed
+    - Creates SSH key pair in memory. Stores the secret key encrypted under `status.id_rsa.enc` using `spec.publicKey`. Stores the public key under `status.id_rsa.pub`
+    - Controller reads `cloudprovider` credentials from seed-shoot namespace
+    - Deploy infrastructure resources
+        - Bastion VM. User data similar to https://github.com/gardener/gardenctl/blob/1e3e5fa1d5603e2161f45046ba7c6b5b4107369e/pkg/cmd/ssh.go#L160-L171. Writes `status.id_rsa.pub` into `authorized_keys` file.
+        - create security groups / firewall rules etc.
+    - Updates status of `Bastion` resource:
+        - With bastion IP under `status.bastionIP`
+        - Sets `status.state` to `Ready` on resource so that the client knows when to initiate the SSH connection
+5. `gardenlet`
+    - Once the `Bastion` resource is in ready state, it syncs back the state to the garden cluster
```

Comment:
Too vague - please be more explicit

Reply:
> - Syncs back the `status.ipAddress` and `status.conditions` of the `Bastion` resource in the seed to the garden cluster in case it changed

@rfranzke is this better?